packages/decoder/src/compose.ts

```diff
@@ -0,0 +1,9 @@
+import type { ParseOptions } from "@effect/schema/AST"
+import * as PR from "@effect/schema/ParseResult"
+
+import type { Decoder } from "@typed/decoder/decoder"
+
+export const compose =
```

Comment:
💭 I definitely won't block on this, but thinking out loud, it'd be nice to adjust these to utilize the `dual` function from Effect like all the standard combinators in Effect